### DISCUSSION TITLE
How to restart DPA-2 training?

### DISCUSSION THREAD

**Discussion by zhbzhbzhbzhb** [deepmodeling/deepmd-kit#3823] [Q&A]
### Summary

How to restart PDA-2 training? 

### DeePMD-kit Version

3.0.0a1.dev82+geed7c8ac

### Backend and its version

torch2.0.0 torchvision 0.15.1 torchaudio-2.0.1

### Details

Should I use: dp train --restart model.ckpt input.json --finetune OpenLAM_2.1.0_27heads_2024Q1.pt --model-branch Domains_OC2M

**Comment by anyangml** [comment]
if you want to resume a finetune experiment from the latest checkpoint, I believe you can simple do 

```
dp train input.json --restart latest_checkpoint.ckpt --skip-neighbor-stat
```

**Reply by JiangXiaoMingSan** [reply]
What is the difference between this command and the restart command when directly training a dp potential function? Do we need to continue providing the large model and -- model branch?

**Reply by anyangml** [reply]
to resume a finetune experiment, you don't have to provide the pretrain model. You only need the latest_finetune_ckpt

**Reply by JiangXiaoMingSan** [reply]
But I used this command to restart an example, which reduced the dataset by some compared to the original one. But after reading the dataset, the program displays as completed. I am using the latest version of deepmdkit (v3.0.0), with the command: dp -- pt train input_finetune. json -- restart model. ckpt-800000. pt -- skip neighbor stat. Once the log file is updated to the following program, it will display as completed.
[2024-12-05 14:40:59,495] DEEPMD INFO    --------------------------------------------------------------------------------------
[2024-12-05 14:40:59,496] DEEPMD INFO    Resuming from model.ckpt-800000.pt.
[2024-12-05 14:40:59,931] DEEPMD INFO    Start to train 300000 steps.
[2024-12-05 14:40:59,941] DEEPMD INFO    Trained model has been saved to: model.ckpt

**Reply by anyangml** [reply]
I am confused.  Is `model.ckpt-800000.pt` the latest ckpt from a crashed finetune experiment or the pretrained model?

**Reply by JiangXiaoMingSan** [reply]
This is the model generated after performing 80W steps of finetune.

**Reply by anyangml** [reply]
then I am not sure why it says `DEEPMD INFO Start to train 300000 steps.` what is the total number of finetune steps in your ` input_finetune. json`

**Reply by JiangXiaoMingSan** [reply]
This model.ckpt-800000. pt is a model generated in 80w steps, and I want it to continue running in 30w steps, so I gave it 30w steps in the input_finetune.json. So the log file says to start training for 300000 steps, but it doesn't continue training and shows as completed. If I want to train a total of 110w steps, should I also write 110w in the input_finetune.json during restart？

**Reply by JiangXiaoMingSan** [reply]
Thank you very much, I have resolved this issue.
